### PR TITLE
Initialize 'Flags' and 'DeclarationModifiers' in constructor of SourceMemberMethodSymbol

### PR DESCRIPTION
**Pull request by AlekseyTs**
This ensures that initialization of these members is not skipped by mistake. The change also:
- Moves 'HasAnyBody' property from SourceMemberMethodSymbol down to SourceOrdinaryMethodSymbol and makes it private. This eliminates the need to define what this concept means for other source method symbols.
- Reduces amount of information passed for synthesized record methods
- Eliminates some steps unnecessary for synthesized record methods.
- Extracts and reuses some common logic.

The second commit moves functions between the files. It might be easier to review commits separately.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review

File: src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs [hunk 1]
```diff
@@ -52,26 +53,35 @@ internal SynthesizedSimpleProgramEntryPointSymbol(SourceMemberContainerTypeSymbo
...
                                   ArrayTypeSymbol.CreateCSharpArray(compilation.Assembly,
                                       TypeWithAnnotations.Create(Binder.GetSpecialType(compilation, SpecialType.System_String, NoLocation.Singleton, diagnostics)))), 0, RefKind.None, "args"));
         }
 
+        private static (DeclarationModifiers, Flags) MakeModifiersAndFlags(SourceMemberContainerTypeSymbol containingType, SingleTypeDeclaration declaration)
+        {
+            bool hasAwait = declaration.HasAwaitExpressions;
+            bool hasReturnWithExpression = declaration.HasReturnWithExpression;
+
+            DeclarationModifiers declarationModifiers = DeclarationModifiers.Static | DeclarationModifiers.Private | (hasAwait ? DeclarationModifiers.Async : DeclarationModifiers.None);
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs:67]: >DeclarationModifiers [](http://example.com/codeflow?start=12&length=20)

nit: var?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Review by 333fred — APPROVED** [review]

**Pull request merged**